cloudinit/net/networkd.py

```diff
@@ -35,6 +36,7 @@ def update_section(self, sec, key, val):
         for k in self.conf_dict.keys():
             if k == sec:
                 self.conf_dict[k].append(key+'='+str(val))
+                self.conf_dict[k] = list(dict.fromkeys(self.conf_dict[k]))
```

Comment:
This line appears to be converting a list to a dict with no values, then converting the keys of the dict back into a list. What does this accomplish?

Reply:
This removes duplicates from a list

Reply:
Ah, right. I'm used to using sets for that. Can you add a comment specifying it removes duplicates? Also, a duplicate in one of the units tests would be helpful. Currently removing this line still allows all tests to pass.